crates/sui-proxy/src/peers.rs

```diff
@@ -193,25 +251,41 @@ impl SuiNodeProvider {
 
             loop {
                 interval.tick().await;
-
-                match Self::get_validators(rpc_url.to_owned()).await {
-                    Ok(summary) => {
-                        let peers = extract(summary);
-                        // maintain the tls acceptor set
-                        let mut allow = nodes.write().unwrap();
-                        allow.clear();
-                        allow.extend(peers);
-                        info!("{} peers managed to make it on the allow list", allow.len());
-                        JSON_RPC_STATE
-                            .with_label_values(&["update_peer_count", "success"])
-                            .inc_by(allow.len() as f64);
-                    }
+                let sui_validators_summary = match Self::get_validators(rpc_url.to_owned()).await {
+                    Ok(summary) => summary,
                     Err(error) => {
                         JSON_RPC_STATE
                             .with_label_values(&["update_peer_count", "failed"])
                             .inc();
-                        error!("unable to refresh peer list: {error}")
+                        error!("unable to refresh peer list: {error}");
+                        continue;
                     }
+                };
+                let bridge_validators_summary =
+                    match Self::get_bridge_validators(rpc_url.to_owned()).await {
+                        Ok(summary) => summary,
+                        Err(error) => {
+                            JSON_RPC_STATE
+                                .with_label_values(&["update_bridge_peer_count", "failed"])
+                                .inc();
+                            error!("unable to refresh sui bridge peer list: {error}");
+                            continue;
+                        }
+                    };
+                // iff both the `sui_validators` and `bridge_validators` lists are up to date, then clear and update the allow list
+                {
+                    let sui_validators = extract(sui_validators_summary);
+                    let bridge_validators = extract_bridge(bridge_validators_summary).await;
+
+                    let mut allow = nodes.write().unwrap();
+                    allow.clear();
+                    allow.extend(sui_validators);
+                    allow.extend(bridge_validators);
+
+                    info!("{} peers managed to make it on the allow list", allow.len());
+                    JSON_RPC_STATE
+                        .with_label_values(&["update_peer_count", "success"])
+                        .inc_by(allow.len() as f64);
                 }
```

Comment:
It's only possible for the set of bridge or sui validators to change once per day (epoch change), while we poll for updates to the system state every 60s. I am comfortable with the risk that if our public rpc goes down for longer than a day, then we won't display metrics for newly joined validators

Reply:
if the service restarts, you're dead in the water in that case.

Reply:
How would you suggest we recover in the case where the service restarts and we're unable to communicate with the backing fullnode?

Reply:
So i don't think we necessarily need to separate these into their own "route handlers", i mean this isn't even a route handler, this is the background task that is responsible for updating the allowed peers, but i do think that this could be improved upon to be a bit more resilient to failures as well as to be a bit clearer about the source of each of the peers.

If the "AllowedPeers" set was able to curry with it a bit more information as to the source of each peer (static, SuiValidator, SuiBridgeNode, WalrusNode, etc) then we could handle the updates to each set independently of each other, and so that a failure to update one set doesn't impact an update to another set. This tagging could be done either by tagging each key, or having a separate set entirely for each type of peer (it doesn't matter which, the implementation is just going to be slightly different). Then we could have the update logic essentially be:

```
loop {
    update_sui_validator_set();
    update_sui_bridge_set();
    update_walrus_validator_set();
    // ... others
}
```

where on each tick we unconditionally try to update each type and a failure in one doesn't impact updates in the others.

Reply:
> the original design with the allowers was to enable the expansion of other routes to accommodate use cases like what john is doing here.

If you're referring to the TLS verifier we have that allows connections based on the `Allower`, then yes John is doing exactly as expected to extend the sets of peers that we allow to be connected. There can only be one allower per network endpoint so unless we have a full complete deployment of this for each "type" (which i don't think is the case), then this is exactly how we should extend it. As far as which http "route" we expect metrics to be pushed to, I think that's a completely orthogonal discussion